src/Parser/Shaman/Restoration/Modules/Spells/Resurgence.js

```diff
@@ -0,0 +1,145 @@
+import HIT_TYPES from 'Parser/Core/HIT_TYPES';
+import Analyzer from 'Parser/Core/Analyzer';
+
+import SpellLink from 'common/SpellLink';
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber, formatPercentage } from 'common/format';
+
+import MasteryEffectiveness from '../Features/MasteryEffectiveness';
+
+class Resurgence extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    masteryEffectiveness: MasteryEffectiveness,
+  }
+
+  maxMana = 1100000;
+  regenedMana = 0;
+  extraMana = 0;
+  resurgence = [];
+  SPELLS_PROCCING_RESURGENCE = {};
+  totalResurgenceGain = 0;
+  hasbottomlessDepths = false;
+  bottomlessDepths = 0;
+
+  on_initialized() {
+    const refreshingCurrentTrait = this.combatants.selected.traitsBySpellId[SPELLS.REFRESHING_CURRENTS.id] || 0;
+    const hasEnergyPendant = this.combatants.selected.hasNeck(ITEMS.STABILIZED_ENERGY_PENDANT.id);
+    this.hasbottomlessDepths = this.combatants.selected.hasTalent(SPELLS.BOTTOMLESS_DEPTHS_TALENT.id);
+
+    // The arcway neck does increase resurgence gained as well
+    if(hasEnergyPendant){
+      this.maxMana *= 1.05;
+    }
+
+    this.SPELLS_PROCCING_RESURGENCE = {
+      [SPELLS.HEALING_SURGE_RESTORATION.id]: 0.006,
+      [SPELLS.HEALING_WAVE.id]: 0.01,
+      [SPELLS.CHAIN_HEAL.id]: refreshingCurrentTrait ? 0.00375 : 0.0025,
+      [SPELLS.UNLEASH_LIFE_TALENT.id]: 0.006,
+      [SPELLS.RIPTIDE.id]: 0.006,
+    };
+  }
+
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+    const isAbilityProccingResurgence = this.SPELLS_PROCCING_RESURGENCE.hasOwnProperty(spellId);
+    const masteryEffectiveness = event.masteryEffectiveness;
+
+    if(!isAbilityProccingResurgence || event.tick){
+      return;
+    }
+
+    if (!this.resurgence[spellId]) {
+      this.resurgence[spellId] = {
+        spellId: spellId,
+        resurgenceTotal: 0,
+        castAmount: 0,
+      };
+    }
+
+    if (event.hitType === HIT_TYPES.CRIT) {
+        this.resurgence[spellId].resurgenceTotal += this.SPELLS_PROCCING_RESURGENCE[spellId] * this.maxMana;
+        this.resurgence[spellId].castAmount += 1;
+    } else if (event.hitType === HIT_TYPES.NORMAL && masteryEffectiveness >= 0.4) {
+      if(this.hasbottomlessDepths) {
+        this.resurgence[spellId].resurgenceTotal += this.SPELLS_PROCCING_RESURGENCE[spellId] * this.maxMana;
+        this.resurgence[spellId].castAmount += 1; 
+      }
+      this.bottomlessDepths += this.SPELLS_PROCCING_RESURGENCE[spellId] * this.maxMana; 
+    }
+  }
+
+  on_toPlayer_energize(event) {
+    const spellId = event.ability.guid;
+
+    if (spellId !== SPELLS.RESURGENCE.id) {     
+      return this.extraMana += event.resourceChange;
+    }
+
+    this.totalResurgenceGain += event.resourceChange;
+  }
+
+  get totalMana() {
+    this.regenedMana = ((this.owner.fightDuration/1000)/5) * 44000;
```

Comment:
Does mana regn. continue while dead? Otherwise might be worth checking (can probably use the death module? haven't looked at it).

Reply:
I've not found a 100% concluding answer on that yet looking through logs - i did not find one where the person didn't gain mana while dead though. but every death that i do find they revive with what seems to be about 36-40k MP5 during their death instead of the expected 44k

Reply:
Alright, should be fine to just assume it continues then I suppose. Argus is probably a bit weird, I dunno if you want to do something specific with that, but I'll just merge and you can add something for Argus later if you think it makes sense (and if it is even detectable?)